### PR TITLE
Enhance select window

### PR DESCRIPTION
**Pull request by divfor**
this is a re-submit for PR #395 - rebase to latest rotmac/master to resolve conflict

### CODE REVIEW

**Comment by divfor** [comment]
Hey @zephraph , I have done this PR and passed all my testing.  Please review for merge.

File: src/Selenium2Library/keywords/_browsermanagement.py [hunk 1]
```diff
@@ -315,7 +323,16 @@ def select_window(self, locator=None):
         | Title Should Be | Popup Title |
         | Select Window |  | | # Chooses the main window again |
         """
-        self._window_manager.select(self._current_browser(), locator)
+        try:
+            return self._current_browser().get_current_window_handle()
```
Comment (just-be-dev) [inline src/Selenium2Library/keywords/_browsermanagement.py:327]: So, I might be missing something. If they provide a locator, couldn't this possible return the wrong window?

**Comment by just-be-dev** [comment]
@emanlove, @pekkaklarck, @ombre42  Can any of you assist in this review?

**Comment by divfor** [comment]
actually the 'return self._current_browser().get_current_window_handle()' is making a result and store it only. after try block the finally block will be executed, at last the function ends and the stored return value is pop up to caller.
I test this and it works well.

**Comment by just-be-dev** [comment]
Overall it looks good to me, but I want to give it another day or two to let someone else look at it before I merge.

**Comment by just-be-dev** [comment]
Hey @divfor, I'm sorry this is taking so long. Being as we haven't heard from anyone else, I'll take some extra time tonight and play around with this until I'm comfortable that I understand exactly everything you're doing. If I can't break it then I'll merge it tomorrow.

**Comment by divfor** [comment]
Thank you, zephraph. 
One more notes: besides my project using it, I also sent this enhancement to some other companies' guys for 2 weeks and did not receive negative feedback.

**Comment by just-be-dev** [comment]
Hey @divfor, I'm planning on releasing 1.7 today. 

Do you want this in that release, or is it okay to wait for 1.7.1? If you want it in today, please merge with master as soon as you're able. 

Thanks!

**Comment by divfor** [comment]
I tried but can not fix the conflicts in CHANGES.rst...
could you help?

**Comment by just-be-dev** [comment]
So we're you able to get it? I can do a PR on your fork if you need.

**Comment by zephraph** [comment]
Well, I opened up another PR that fixed the merge issue... but you're still doing work?

Is this issue not ready to be merged?

**Comment by just-be-dev** [comment]
Tell me when you're done making changes and I'll open a PR on your branch to do the merge.

**Comment by divfor** [comment]
I still cannot fix the conflicts. please start from the latest 'Update CHANGES.rst' -- aa50ade.

**Comment by just-be-dev** [comment]
Hopefully this builds successfully! I'm really ready to push out this release...

File: src/Selenium2Library/locators/windowmanager.py [hunk 2]
```diff
@@ -53,21 +61,36 @@ def _select_by_url(self, browser, criteria):
...
+            browser.switch_to_window(handles[0])
             return
-
-        try:
-            self._select_by_name(browser, criteria)
-            return
-        except ValueError: pass
-
+        for handle in browser.get_window_handles():
+            browser.switch_to_window(handle)
```
Comment (just-be-dev) [inline src/Selenium2Library/locators/windowmanager.py:72]: Wouldn't it be better to only switch to the window when you know that it meets the right criteria? If it's unable to locate the handle you want to switch to, I would think the expected behavior would be to stay on the current handle.

File: src/Selenium2Library/locators/windowmanager.py [hunk 3]
```diff
@@ -53,21 +61,36 @@ def _select_by_url(self, browser, criteria):
...
+            if criteria == handle:
+                return
```
Comment (just-be-dev) [inline src/Selenium2Library/locators/windowmanager.py:74]: As a continuation of the above comment, this part would be

``` python
if criteria == handle:
    browser.switch_to_window(handle)
    return
```

File: src/Selenium2Library/locators/windowmanager.py [hunk 4]
```diff
@@ -53,21 +61,36 @@ def _select_by_url(self, browser, criteria):
...
+            for item in browser.get_current_window_info()[2:4]:
+                if item.strip().lower() == criteria.lower():
+                    return
```
Comment (just-be-dev) [inline src/Selenium2Library/locators/windowmanager.py:77]: This would also call `bowser.switch_to_handle` instead of just returning.

**Comment by divfor** [comment]
Yes, we can add it like what _select_matching() does.
I ignored it because a VauleError exception must be raised to fail this keyword if no match window found

**Pull request merged**